network/network_linux.go

```diff
@@ -345,23 +353,35 @@ func applyDnsConfig(extIf *externalInterface, ifName string) error {
 				continue
 			}
 
-			buf := fmt.Sprintf("--set-dns=%s", server)
-			setDnsList = setDnsList + " " + buf
+			setDnsList = setDnsList + " " + server
```

Comment:
is it expected for `extIf.DNSInfo.Servers` to have >1 item? I think systemd-resolve will fail the way the commands are currently written with this "setDnsList".

it seems like `resolvectl` can handle this:
```bash
resolvectl help
[...]
  dns [LINK [SERVER...]]       Get/set per-interface DNS server address
```
`$ resolvectl dns azure0 192.168.1.1 192.168.1.2` should work.


but it doesn't look like `systemd-resolve` can, and it only expects a single address when invoked like this:
```bash
systemd-resolve help
[...]
     --set-dns=SERVER       Set per-interface DNS server address
```
`$ systemd-resolve --interface=azure0 --set-dns=192.168.1.1 192.168.1.2` definitely will not work.

cc @tamilmani1989 can you clarify above?